src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_grand_champions.cpp

```diff
@@ -355,47 +697,221 @@ class boss_warrior_toc5 : public CreatureScript
+                uiPhase = 3;
+                uiPhaseTimer = spell->GetDuration();
+            }
+        }
+
+        void MovementInform(uint32 type, uint32 id) override
+        {
+            if (type != POINT_MOTION_TYPE || !LookingForMount)
                 return;
 
-            uiPhaseTimer = 15000;
-            uiPhase = 1;
+            if (id == 0)
+            {
+                // Reached point, mounting again
+                if (Creature* mount = FindMyMount(me))
+                {
+                    if (!me->IsWithinDist(mount, INTERACTION_DISTANCE))
+                        return;
+
+                    LookingForMount = false;
+                    me->ApplySpellImmune(SPELL_TRAMPLE_AURA, IMMUNITY_ID, SPELL_TRAMPLE_AURA, true); // Champion should be immune to trample here
+                    mount->SetOrientation(4.714f); // looking towards arena
+                    me->GetMotionMaster()->Clear();
+                    me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE); // to prevent boss from bugging out
+                    me->SetHealth(me->GetMaxHealth());
+                    me->setRegeneratingHealth(true);
+                    mount->SetVisible(true);
+                    me->EnterVehicle(mount, SEAT_ID_0);
+
+                    // In rare scenario all players can die meanwhile one of the champions is going for new mount, and then the champion ends in a bugged state
+                    // so we must check do we have alive players in the map first, else we call evade
+                    bool foundPlr = false;
+                    Map::PlayerList const& players = me->GetMap()->GetPlayers();
+                    if (me->GetMap()->IsDungeon() && !players.isEmpty())
+                    {
+                        for (Map::PlayerList::const_iterator itr = players.begin(); itr != players.end(); ++itr)
+                        {
+                            Player* player = itr->GetSource();
+                            if (player && !player->IsGameMaster() && player->IsAlive())
+                            {
+                                foundPlr = true;
+                                break;
+                            }
+                        }
+                    }
+                    if (foundPlr)
+                    {
+                        uiPhase = 2;
+                        uiPhaseTimer = 2000; // slight delay before attacking
+                    }
+                    else
+                        SetGrandChampionToEvadeMode(me);
+                }
+            }
+        }
 
-            bHome = false;
+        void DamageTaken(Unit* /*attacker*/, uint32& damage) override
+        {
+            if (damage >= me->GetHealth() && me->GetVehicleBase())
+            {
+                damage = 0;
+                me->ApplySpellImmune(SPELL_TRAMPLE_AURA, IMMUNITY_ID, SPELL_TRAMPLE_AURA, true); // Champion should be immune to trample here
+                me->ExitVehicle(&me->GetPosition());
```

Comment:
... in addition, `(&me->GetPosition())` on this exact line crashes the Travis build.